### PR TITLE
Fixed directory ownership handling and allowed node awareness to be read from an env variable

### PR DESCRIPTION
**Pull request by dn0**
Related to #70 and #71

- Setting ownership of the `/elasticsearch` directory is now part of the
  Dockerfile. Ownership of the `/data` folder is changed only when
  started for the first time.
- The `SHARD_ALLOCATION_AWARENESS_ATTR` environment variable was
  previously required to contain a file path. Now, the environment
  variable can contain the actual value of the node metadata
  attribute. Also, this attribute can be set on any node.

### CODE REVIEW

**Comment by dn0** [comment]
Not tested - do not merge yet.

**Comment by dn0** [comment]
@pires I've tested this change together with ES version 6.4.2 and did not observe any problems.
Also, this change moves the `export NODE_NAME` after the "SHARD_ALLOCATION_AWARENESS" code block, which means that my node names are now prefixed with allocation awareness attribute (which I assume was the original intention).

**Review by pires — CHANGES_REQUESTED** [review]
Please, resolve conflicts.

**Comment by dn0** [comment]
@pires Done. Please review carefully because after merging master I didn't deploy it yet.

**Comment by bambrozio** [comment]
Reviewing...

**Review by bambrozio — APPROVED** [review]
LGTM.

Test results:

 - Image Successfully built:
```
$ git clone https://github.com/pires/docker-elasticsearch.git
Cloning into 'docker-elasticsearch'...
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 606 (delta 0), reused 3 (delta 0), pack-reused 599
Receiving objects: 100% (606/606), 145.98 KiB | 165.00 KiB/s, done.
Resolving deltas: 100% (274/274), done.

$ cd docker-elasticsearch/

$ git fetch origin pull/75/head:reviewing75
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 8 (delta 4), reused 8 (delta 4), pack-reused 0
Unpacking objects: 100% (8/8), done.
From https://github.com/pires/docker-elasticsearch
 * [new ref]         refs/pull/75/head -> reviewing75

$ git checkout reviewing75
Switched to branch 'reviewing75'

$ docker build --no-cache -t quay.io/pires/docker-elasticsearch:6.4.2 .
(...)
Successfully built b0e7570736ae
Successfully tagged quay.io/pires/docker-elasticsearch:6.4.2
```

And successfully deployed by using `pires/kubernetes-elasticsearch-cluster` (and helm chart):
```
$ kubectl get pod -w
NAME                        READY     STATUS    RESTARTS   AGE
mw-elasticsearch-data-0     1/1       Running   0          1m
mw-elasticsearch-ingest-0   1/1       Running   0          1m
mw-elasticsearch-master-0   1/1       Running   0          1m

$ kubectl exec -it mw-elasticsearch-ingest-0 -- env | grep ES_VERSION
ES_VERSION=6.4.2
$ kubectl exec -it mw-elasticsearch-master-0 -- env | grep ES_VERSION
ES_VERSION=6.4.2
$ kubectl exec -it mw-elasticsearch-data-0 -- env | grep ES_VERSION
ES_VERSION=6.4.2

$ kubectl logs mw-elasticsearch-master-0 | grep -i started
[2018-10-16T09:46:25,813][INFO ][o.e.n.Node               ] [mw-elasticsearch-master-0] started

$ kubectl logs mw-elasticsearch-data-0 | grep -i started
[2018-10-16T09:46:43,400][INFO ][o.e.n.Node               ] [mw-elasticsearch-data-0] started

$ kubectl logs mw-elasticsearch-ingest-0 | grep -i started
[2018-10-16T09:46:41,085][INFO ][o.e.n.Node               ] [mw-elasticsearch-ingest-0] started

$ kubectl logs mw-elasticsearch-master-0 | grep -i added
[2018-10-16T09:46:38,290][INFO ][o.e.c.s.MasterService    ] [mw-elasticsearch-master-0] zen-disco-node-join, reason: added {{mw-elasticsearch-ingest-0}...
[2018-10-16T09:46:41,097][INFO ][o.e.c.s.MasterService    ] [mw-elasticsearch-master-0] zen-disco-node-join, reason: added {{mw-elasticsearch-data-0}...
```

**Comment by pires** [comment]
Thank you very much!

**Pull request closed without merging**
